File: FitTrack.API/FitTrack.API/Models/Purchase.cs
namespace FitTrack.API.Models;
public class Purchase
{
    public int Id { get; set; } 
    public required int ItemId { get; set; }      
    public required ItemType ItemType { get; set; }
    public required DateTime Date { get; set; }  
    public required int Quantity { get; set; }
    
    public required string UserId { get; set; }
    public required User User { get; set; }
    
    public Good? Good { get; set; }
    public Service? Service { get; set; }
}

public enum ItemType
{
    Good,
    Service
}

File: fittrack_mobile_app/lib/models/purchase.dart
import 'user.dart';

class Purchase {
  final int id;
  final int itemId;
  final int userId;
  final DateTime date;
  final int quantity;
  final ItemType itemType;
  final User user;

  Purchase({
    required this.id,
    required this.itemId,
    required this.userId,
    required this.date,
    required this.quantity,
    required this.itemType,
    required this.user,
  });

  factory Purchase.fromJson(Map<String, dynamic> json) {
    return Purchase(
      id: json['id'],
      itemId: json['itemId'],
      userId: json['userId'],
      date: DateTime.parse(json['date']),
      quantity: json['quantity'],
      itemType: ItemType.values[json['itemType']],
      user: User.fromJson(json['user']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'itemId': itemId,
      'userId': userId,
      'date': date.toIso8601String(),
      'quantity': quantity,
      'itemType': itemType.index,
      'user': user.toJson(),
    };
  }
}

enum ItemType {
  Good,
  Service,
}
